sys/vm/vm_map.c

```diff
@@ -4544,8 +4704,16 @@ vm_map_remove_locked(vm_map_t map, vm_offset_t start, vm_offset_t end)
 	}
 
 	result = vm_map_delete(map, start, end, true);
-	if (vm_map_reservation_is_unmapped(map, reservation))
-		vm_map_reservation_delete_locked(map, reservation);
+	if (vm_map_reservation_is_unmapped(map, reservation)) {
+#ifdef CHERI_CAPREVOKE
+		if (quarantine_unmapped_reservations &&
+		    start < VM_MAXUSER_ADDRESS) {
```

Comment:
Checking `start < VM_MAXUSER_ADDRESS` seems a bit weird to me - shouldn't we just avoid fiddling with `kernel_map`?

Reply:
We do use reservations in the purecap kernel map (and submaps) and until we have a revoker we'd be leaking VA didn't have some sort of check. I'm not sure if this is the right one though.